examples/ota_from_flash.rs

```diff
@@ -0,0 +1,35 @@
+extern crate esp_ota;
+use log::*;
+
+const FIRMWARE_ID: &str = "alpha";
+
+const NEW_APP: &[u8] = include_bytes!("../beta.bin");
+
+fn main() {
+    esp_idf_svc::sys::link_patches();
+    esp_idf_svc::log::EspLogger::initialize_default();
+
+    info!("FIRMWARE ID : {}", FIRMWARE_ID);
+    if FIRMWARE_ID == "alpha" {
+        let mut ota = esp_ota::OtaUpdate::begin().unwrap();
+        // write app to flash
+
+        for app_chunk in NEW_APP.chunks(4096) {
+            if let Err(err) = ota.write(app_chunk) {
+                error!("Failed to write chunk");
+                break;
```

Comment:
This will just exit the `for` loop and go directly to `ota.finalize()`. Here we know the update failed already, and I don't think it's the best course of action to try and finalize the update. I think it's better in general (and teach the readers) to exit with an error here. Only run the finalize code if all write calls succeeded.